### PR TITLE
remove unnecessary block from bound node for single line lambda

### PR DESCRIPTION
**Pull request by heejaechang**
**Customer scenario**

it affects diagnostic analyzer author not VS users. for the author, when they get called for IAnonymousFunctionExpression, they might see 2 nested IBlock operations instead of 1.

**Bugs this fixes:**

https://github.com/dotnet/roslyn/issues/22196
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems?id=507638

**Workarounds, if any**

there is no workaround.

**Risk**

this is new API, no risk.

**Performance impact**

there is no perf risk

**Is this a regression from a previous update?**

No

**Root cause analysis:**

We wrapped statements on block twice unnecessarily. 

**How was the bug found?**

dogfooding and testing.

### CODE REVIEW

**Comment by heejaechang** [comment]
@AlekseyTs @cston @jcouv @dotnet/analyzer-ioperation can you take a look?

**Comment by heejaechang** [comment]
ping?

File: src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests.vb [hunk 1]
```diff
@@ -702,6 +702,141 @@ ICatchClause (Exception type: System.Exception) (OperationKind.CatchClause) (Syn
...
+Class Test
+    Sub Method()
+        Dim a As Action = Sub() Console.Write("hello")'BIND:"Sub() Console.Write("hello")"
+    End Sub
+End Class]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IAnonymousFunctionExpression (Symbol: Sub ()) (OperationKind.AnonymousFunctionExpression, Type: null) (Syntax: 'Sub() Conso ... te("hello")')
+  IBlockStatement (3 statements) (OperationKind.BlockStatement, IsImplicit) (Syntax: 'Sub() Conso ... te("hello")')
+    IExpressionStatement (OperationKind.ExpressionStatement) (Syntax: 'Console.Write("hello")')
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests.vb]: >IExpressionStatement [](start = 4, length = 20)

Should this node be Implicit?

Comment (heejaechang) [reply src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests.vb]: spec says sub expects statement?

(https://docs.microsoft.com/en-us/dotnet/visual-basic/reference/language-specification/expressions)
```
SingleLineLambda
    : LambdaModifier* 'Function' ( OpenParenthesis ParameterList? CloseParenthesis )? Expression
    | 'Sub' ( OpenParenthesis ParameterList? CloseParenthesis )? Statement
    ;
```
so IExpressionStatement is valid there?

Comment (heejaechang) [reply src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests.vb]: tagging @AlekseyTs 

also, syntax does have expression statement there.

image

**Comment by heejaechang** [comment]
ping? @jinujoseph approval?

**Comment by MattGertz** [comment]
For Escrow -- please create a VSO bug.

**Comment by jinujoseph** [comment]
I have added VSO bug https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems?id=507638

**Comment by heejaechang** [comment]
@jinujoseph done.

**Comment by heejaechang** [comment]
still pending?

**Comment by jinujoseph** [comment]
yes , this is still pending approval

**Comment by natidea** [comment]
Approved to merge via [507638](https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems/edit/507638)

**Comment by jinujoseph** [comment]
test windows_debug_unit64_prtest

**Pull request merged**